From c9ad8bb5f819eb58fba7d43f30195706a58b8f26 Mon Sep 17 00:00:00 2001
Message-ID: <c9ad8bb5f819eb58fba7d43f30195706a58b8f26.1718966788.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1718966788.git.sharlatanus@gmail.com>
References: <cover.1718966788.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 14 Jun 2024 07:56:41 +0100
Subject: [PATCH 23/30] gnu: Add python-fast-histogram.

* gnu/packages/python-science.scm (python-fast-histogram): New variable.

Change-Id: Ibecdd87db9c132052a8eca76c01108c4f6e624c4
---
 gnu/packages/python-science.scm | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/gnu/packages/python-science.scm b/gnu/packages/python-science.scm
index 401beecbcd..931f3c57d3 100644
--- a/gnu/packages/python-science.scm
+++ b/gnu/packages/python-science.scm
@@ -1519,6 +1519,38 @@ (define-public python-baycomp
 higher scores.")
     (license license:expat)))
 
+(define-public python-fast-histogram
+  (package
+    (name "python-fast-histogram")
+    (version "0.14")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "fast_histogram" version))
+       (sha256
+        (base32 "1sk9xa85cgm4sylzblwv3qr2dmm0ic06zkwxqa2xlazjiawp629r"))))
+    (build-system pyproject-build-system)
+    (arguments
+     (list
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-before 'check 'build-extensions
+            (lambda _
+              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
+    (propagated-inputs (list python-numpy))
+    (native-inputs (list python-hypothesis python-pytest))
+    (home-page "https://github.com/astrofrog/fast-histogram")
+    (synopsis "Fast simple 1D and 2D histograms")
+    (description
+     "The fast-histogram mini-package aims to provide simple and fast
+histogram functions for regular bins that don't compromise on performance. It
+doesn't do anything complicated - it just implements a simple histogram
+algorithm in C and keeps it simple. The aim is to have functions that are fast
+but also robust and reliable. The result is a 1D histogram function here that
+is 7-15x faster than @code{numpy.histogram}, and a 2D histogram function that
+is 20-25x faster than @code{numpy.histogram2d}.")
+    (license license:bsd-3)))
+
 (define-public python-fastcluster
   (package
     (name "python-fastcluster")
-- 
2.41.0

